src/search.c

```diff
@@ -1669,207 +1648,84 @@ search_find_in_files(const gchar *utf8_search_text, const gchar *dir, const gcha
 	if (search_text == NULL)
 		search_text = g_strdup(utf8_search_text);
 
-	/* set grep command and options */
-	argv_prefix = g_new0(gchar*, 1 + opts_argv_len + 3 + 1);	/* last +1 for recursive arg */
-
-	argv_prefix[0] = command_grep;
-	for (i = 0; i < opts_argv_len; i++)
-	{
-		argv_prefix[i + 1] = g_strdup(opts_argv[i]);
-	}
-	g_strfreev(opts_argv);
-
-	i++;	/* correct for tool_prefs.grep_cmd */
-	argv_prefix[i++] = g_strdup("--");
-	argv_prefix[i++] = search_text;
-
-	/* finally add the arguments(files to be searched) */
-	if (strstr(argv_prefix[1], "r"))	/* recursive option set */
-	{
-		/* Use '.' so we get relative paths in the output */
-		argv_prefix[i++] = g_strdup(".");
-		argv_prefix[i++] = NULL;
-		argv = argv_prefix;
-	}
-	else
-	{
-		argv_prefix[i++] = NULL;
-		argv = search_get_argv((const gchar**)argv_prefix, dir);
-		g_strfreev(argv_prefix);
-	}
-
-	if (argv == NULL)	/* no files */
-	{
-		return FALSE;
-	}
+	argv[0] = g_strdup("--");
+	argv[1] = search_text;
+	argv[2] = ".";
```

Comment:
This is a literal, and the `g_strfreev()` call below will try to free it -- aka boom.